docs/upgrade/canary-upgrade.md

```diff
@@ -0,0 +1,371 @@
+
+   ```bash
+   $ helm install istio-operator-v17x --create-namespace --namespace=istio-system --set-string operator.image.tag=0.7.1 banzaicloud-stable/istio-operator
+   ```
+
+1. Apply an `Istio` Custom Resource and let the operator reconcile the Istio 1.7 control plane.
+
+   ```bash
+   $ kubectl apply -n istio-system -f https://raw.githubusercontent.com/banzaicloud/istio-operator/release-1.7/config/samples/istio_v1beta1_istio.yaml
+   ```
+
+1. Check that the Istio 1.7 control plane is also deployed.
+
+   ```bash
+   $ kubectl get po -n=istio-system
+   NAME                                        READY   STATUS    RESTARTS   AGE
+   istio-ingressgateway-55b89d99d7-4m884       1/1     Running   0          6m38s
+   istio-operator-v16x-0                       2/2     Running   0          7m18s
+   istio-operator-v17x-0                       2/2     Running   0          76s
+   istiod-676fc6d449-9jwfj                     1/1     Running   0          10s
+   istiod-istio-sample-v17x-7dbdf4f9fc-bfxhl   1/1     Running   0          18s
+   ```
+
+### Migrate data plane
+
+#### Migrate ingress
+
+1. Move the ingress gateway so as to utilize the new Istio 1.7 control plane.
+
+   ```bash
+   $ kubectl patch mgw -n istio-system istio-ingressgateway --type=json -p='[{"op": "replace", "path": "/spec/istioControlPlane/name", "value": "istio-sample-v17x"}]'
+   ```
+
+#### Migrate first namespace
+
+1. Label the first namespace for the new control plane.
+
+   ```bash
+   $ kubectl label ns demo-a istio-injection- istio.io/rev=istio-sample-v17x.istio-system
+   ```
+
+   The new `istio.io/rev` label needs to be used for the new revisioned control planes to be able to perform sidecar injection.
+   The `istio-injection` label must be removed because it takes precedence over the `istio.io/rev` label for backward compatibility.
+
+1. Make sure that the labeling is correct.
+
+   ```bash
+   $ kubectl get ns demo-a -L istio-injection -L istio.io/rev
+   NAME     STATUS   AGE   ISTIO-INJECTION   REV
+   demo-a   Active   12m                     istio-sample-v17x.istio-system
+   ```
+
+1. Restart the pod in the namespace.
+
+   ```bash
+   $ kubectl rollout restart deployment -n demo-a
+   ```
+
+1. Make sure that now the new 1.7 sidecar proxy is used for the new pod.
+
+   ```bash
+   $ APP_A_POD_NAME=$(kubectl get pods -n demo-a -l k8s-app=app-a -o=jsonpath='{.items[0].metadata.name}')
+   $ kubectl get po -n=demo-a $APP_A_POD_NAME -o yaml | grep istio/proxyv2:
+       image: docker.io/istio/proxyv2:1.7.0
+       image: docker.io/istio/proxyv2:1.7.0
+       image: docker.io/istio/proxyv2:1.7.0
+       image: docker.io/istio/proxyv2:1.7.0
+   ```
+
+#### Test communication
+
+Let's make sure that encrypted communication still works between pods that use different control planes.
+This is a very important part and the reason why the data plane migration can be done gradually, because the communication works even between pods on different control planes.
```

Comment:
```suggestion
The reason why the data plane migration can be done gradually is that the communication works even between pods on different control planes. In this step please it will be verified that the communication works in such a way, making the upgrade flow a safe one.
```